debezium-core/src/test/java/io/debezium/jdbc/JdbcValueConvertersTest.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.jdbc;
+
+import java.sql.Types;
+
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.debezium.doc.FixFor;
+import io.debezium.relational.Column;
+import io.debezium.relational.ColumnEditor;
+
+public class JdbcValueConvertersTest {
+    private ColumnEditor columnEditor;
+
+    @Before
+    public void beforeEach() {
+        columnEditor = Column.editor();
+    }
+
+    @Test
+    @FixFor("DBZ-3989")
+    public void testConvertInteger() {
+        Column column = columnEditor.name("col1").type("INTEGER").jdbcType(Types.INTEGER).defaultValue("1 ").create();
+        Field field = new Field("test", -1, Schema.INT32_SCHEMA);
+        JdbcValueConverters converters = new JdbcValueConverters();
+
+        Object convertedData = converters.convertInteger(column, field, "1 ");
```

Comment:
What happens with Strings like `abcABC `?

Reply:
Actually this trim only applies on convert numerical stuff

Reply:
Ah, then I was just irritated by `if (data instanceof String) {`. Never mind.